phalcon/di/service/serviceregistry.zep

```diff
@@ -0,0 +1,115 @@
+namespace Phalcon\Di\Service;
+
+use Phalcon\DiInterface;
+use Phalcon\Di\Exception;
+use Phalcon\Di\ServiceInterface;
+use Phalcon\Config\Adapter\Yaml;
+
+/**
+ * Phalcon\Di\Service\ServiceRegistry
+ *
+ **/
+class ServiceRegistry
+{
+
+	/**
+	 *
+	 */
+	protected di;
+
+	/**
+	 *
+	 */
+	public function __construct(<DiInterface> di)
+	{
+		let this->di = di;
+	}
+
+	/**
+	 *
+	 */
+	public function registerDir(dirPath)
+	{
+		var dirPathItem;
+
+		switch (gettype(dirPath)) {
```

Comment:
```zep
switch typeof dirPath {
    // ...
}
```

Reply:
I like that.  I did this without Internet access or Zephir documentation.